### ISSUE TITLE
Why "integer" instead of "int" in the primitive schemas?

### ISSUE THREAD

**Issue by jbkoh** [Azure/opendigitaltwins-dtdl#173]
Hi there,

In my understanding of the context file, it seems DTDL adopts XSD vocabularies. However, the definition of "integer" is mismatched between the DTDL's documentation vs XSD's definition.

### DTDL v3:
- integer
> a signed integral numeric value that is expressible in 4 bytes

### XSD
- [int](https://www.w3.org/TR/xmlschema-2/#int): 
> [Definition:]   int is [·derived·](https://www.w3.org/TR/xmlschema-2/#dt-derived) from [long](https://www.w3.org/TR/xmlschema-2/#long) by setting the value of [·maxInclusive·](https://www.w3.org/TR/xmlschema-2/#dt-maxInclusive) to be 2147483647 and [·minInclusive·](https://www.w3.org/TR/xmlschema-2/#dt-minInclusive) to be -2147483648. The [·base type·](https://www.w3.org/TR/xmlschema-2/#dt-basetype) of int is [long](https://www.w3.org/TR/xmlschema-2/#long).

- [integer](https://www.w3.org/TR/xmlschema-2/#integer): 
> [Definition:]   integer is [·derived·](https://www.w3.org/TR/xmlschema-2/#dt-derived) from [decimal](https://www.w3.org/TR/xmlschema-2/#decimal) by fixing the value of [·fractionDigits·](https://www.w3.org/TR/xmlschema-2/#dt-fractionDigits) to be 0and disallowing the trailing decimal point. This results in the standard mathematical concept of the integer numbers. The [·value space·](https://www.w3.org/TR/xmlschema-2/#dt-value-space) of integer is the infinite set {...,-2,-1,0,1,2,...}. The [·base type·](https://www.w3.org/TR/xmlschema-2/#dt-basetype) of integer is [decimal](https://www.w3.org/TR/xmlschema-2/#decimal).

like the hierarchy below
![image](https://github.com/Azure/opendigitaltwins-dtdl/assets/1572627/19838238-fbff-4d88-b8b5-8cca65a03002)

Basically, `xsd:integer` is unbounded and `xsd:int` is the 32bits integer as in your definition

I know I'm a bit pedantic, but wonder if there can be a chance to deprecate integer and start using int for long term alignment with one of the common standards.

**Comment by jrdouceur** [comment]
I don't know the history or the specific considerations that went into DTDL's choices for primitive schema terms, but I can make a few observations.  First, DTDL prefers full words such as "integer" and "boolean" over truncations like "int" and "bool", but this is a lexical preference that does not drive the definition of the terms.  Second, although the DTDL schemas make reference to XSD for some definitional aspects, they also refer directly to other standards including IEEE 754, ISO 8601, and IETF RFC 3339.  Third, "integer" is not the only schema for which DTDL and XSD are not in alignment:

* "duration" -- XSD allows prefixing with a minus sign and does not allow any forms with the "W" designator.
* "string" -- XSD explicitly disclaims how UCS characters are encoded, whereas DTDL specifies UTF-8.

Some current and upcoming applications of DTDL relate to other standards, such as [AVRO](https://avro.apache.org/) and [Protobuf](https://protobuf.dev/).  I expect that if there are any future changes to DTDL schema definitions, these will be driven mainly by interop issues rather than by lexical alignment with XSD or the nomenclature of any other standard.

**Comment by koppor** [comment]
I miss the relation to related work.

- RFC 8428 (Sensor Measurement Lists (SenML)) -> https://datatracker.ietf.org/doc/html/rfc8428#section-12.1
- SensdorML -> https://www.ogc.org/standard/sensorml/
- QUDT -> https://www.qudt.org/
  - Slide deck: https://github.com/qudt/qudt-public-repo/blob/main/docs/2020-04-28%20Intro%20to%20QUDT.pdf
  - Units: https://www.qudt.org/doc/DOC_VOCAB-UNITS.html
  - Number of units: 1522
  - There are 746 quantitykinds
   - There are 148 dimension vectors

Especially, currently, I miss the physical unit. A temperature of 40.0°C is different from 40.0°F - even though the double value is the same...

---

SenML encodes temparature like that:

```json
[
  {"n":"urn:dev:ow:10e2073a01080063","u":"Cel","v":23.1}
]
```

---

Using QUDT, a measurement would be encoded as follows in JSON:

```json
{
  "value": 40,
  "unit": "http://qudt.org/vocab/unit/Celsius",
  "quantityKind": "http://qudt.org/vocab/quantitykind/Temperature",
  "description": "Measurement of temperature"
}
```

Reads nice.

---

Historical note. A decade ago, "everyone" talked about IoT. See for instance https://www.postscapes.com/internet-of-things-protocols/. Most of the lings are dead now... Nevertheless, I think, should be no re-invention of the wheel.

**Comment by koppor** [comment]
Another thing to consider is "Eclipse Vorto", which is a language for digital twins - governede by the Eclipse Foundation, where Microsoft is also involved: https://eclipse.dev/vorto/
